### ISSUE TITLE
Interactive Tour of Growing Guides

### ISSUE THREAD

**Issue by roryaronson** [openfarmcc/OpenFarm#540]
Something like this could help new users learn to use OpenFarm. We could integrate this into the onboarding flow, show an alert to non-logged-in users encouraging them to take it, and also place a link to it at the bottom of the 'See how it works' section on the homepage.

[Foundation Joyride](http://foundation.zurb.com/docs/components/joyride.html) is an implementation option. Maybe there are others, too. Thoughts?

![guide with tutorial](https://cloud.githubusercontent.com/assets/5524043/6593032/98699042-c791-11e4-9c68-0b302047a2be.png)

**Comment by simonv3** [comment]
I feel like industry consensus is - if you need this your thing is too complicated.

But we're dealing with something fairly complex here, so I dunno.

**Comment by roryaronson** [comment]
I feel ya. Though I still see value in this to jumpstart users, and convey more information to them than they may pick up on their own.

I mean, even Slack has a tour when you first sign in. Its simple and straightforward for a simple and straightforward thing, but it was still useful to teach me things I probably wouldn't have found out otherwise like how to mention @channel, or how DMs and privacy with admins works, or how I can tweak my notifications.

Many apps today will do a feature release and have a mini tour of sorts that highlights the new stuff.

I think that often, something like a blatant tour is used as a corner cut and replacement for designing the actual product well. Which I think we should absolutely avoid, and still strive to do great design in all aspects.

Do you have any reading on this you can share? I'd love to learn more and keep the conversation going.

**Comment by warpling** [comment]
I agree strongly with ya both.No specific articles come to mind for me, but I think the general design principal speaks for itself. In situations like this where things are more complex I think the ideal way to teach is a progressive reveal as you explore the app. That said, things like that can get complicated fast, and the quick solution would be something closer to an onboarding plug in. Here’s a [new one](http://github.hubspot.com/shepherd/docs/welcome/)

 I just saw recently, but of course there are dozens and dozens more.


~Ryan

**Comment by simonv3** [comment]
So Samuel from useronboard.com has [this to say](http://www.useronboard.com/advice/#comment-1705175945):) and [this](http://www.useronboard.com/advice/#comment-1881025229).

Then there's [this article](http://www.smashingmagazine.com/2014/11/20/refining-your-mobile-onboarding-experience-using-visual-analytics/) on smashing magazine.

Here's some [other patterns](http://www.pttrns.com/categories/21-coach-marks) for coach marks

I think what might work here is a reveal as you scroll - unobtrusively in the same style as the alerts (so that an alert is consistent and a user knows what it means). 

@warpling I wonder if it's possible to interact with the tour in ways that aren't standard - like on scroll, or when something gets revealed. Would be interesting to explore further.

**Comment by roryaronson** [comment]
Thanks for the links Simon. I would be really interested to interview someone myself and watch them use the site, or pay someone who has never used the site to do something like what Samuel does.

I think the main difference between OpenFarm and most other apps/sites is that 1) it is pretty complex  with a lot of ways to use it and 2) (I think) we want our users to know how everything is working. eg: how does compatibility work? Why are Guides authored by a single person? which I think requires some explicit teaching moments.

I imagine the Interactive Tour to be opt-in, never forced on the user. So either they have decided to take a tour from the homepage, or we show an alert that prompts new users to take a tour. I think forced tours are annoying, but an opted-in one can go a long way towards getting buy-in from the person to actually learn about OpenFarm.

I'd love to whip up a #yolo prototype tour and have some people try it out as an experiment.
